### PR TITLE
Python 3.x support

### PR DESCRIPTION
**Pull request by lheckemann**
So far, only applied 2to3. This already fixes installing. We'll be testing this soon.

### CODE REVIEW

**Comment by globin** [comment]
@fredreichbier @cornelinux In my eyes this should be ready.

All the tests run successfully and it has been tested actively, too.

**Comment by cornelinux** [comment]
@plettich should we simply drop python2 support and move to 3?

File: pi_ldapproxy/proxy.py [hunk 1]
```diff
@@ -3,15 +3,16 @@
...
+import urllib.request, urllib.parse, urllib.error
+from io import BytesIO
 from functools import partial
 
 from ldaptor.protocols import pureldap
 from ldaptor.protocols.ldap import ldaperrors
 from ldaptor.protocols.ldap.ldapclient import LDAPClient
 from ldaptor.protocols.ldap.ldapconnector import connectToLDAPEndpoint
 from ldaptor.protocols.ldap.proxybase import ProxyBase
+from six import ensure_str
```
Comment (cornelinux) [inline pi_ldapproxy/proxy.py:15]: This does not work out - probably with the six-version we are using in requirements.txt?
We should then also update six to 1.16  in the requirements.txt

~~~
builtins.ImportError: cannot import name 'ensure_str' from 'six' (.../ldapproxy/lib/python3.8/site-packages/six.py)
~~~

**Review by cornelinux — CHANGES_REQUESTED** [review]

File: pi_ldapproxy/proxy.py (hunk 1 above)
Comment (cornelinux) [reply pi_ldapproxy/proxy.py:15]: OK, I will update the reqs on my own.

**Pull request merged**